diff --git a/lib/tarballs/build.js b/lib/tarballs/build.js
index 6dc6c060ccb11346ba9e9d38346521e2fd0b9e77..515aa367c767d7c5434c317a53057ded43873c26 100644
--- a/lib/tarballs/build.js
+++ b/lib/tarballs/build.js
@@ -51,8 +51,14 @@ async function build(c, options = {}) {
     const addDependencies = async () => {
         const yarnRoot = findYarnWorkspaceRoot(c.root) || c.root;
         if (fs.existsSync(path.join(yarnRoot, 'yarn.lock'))) {
-            await fs.copy(path.join(yarnRoot, 'yarn.lock'), path.join(c.workspace(), 'yarn.lock'));
-            await exec('yarn --no-progress --production --non-interactive', { cwd: c.workspace() });
+            const yarnVersion = (await exec('yarn -v')).stdout.charAt(0)
+            console.log(yarnVersion)
+            if (yarnVersion === '1') {
+              await fs.copy(path.join(yarnRoot, 'yarn.lock'), path.join(c.workspace(), 'yarn.lock'))
+              await exec('yarn --no-progress --production --non-interactive', {cwd: c.workspace()})
+            } else {
+              await exec('yarn workspaces focus --production', {cwd: yarnRoot})
+            }
         }
         else {
             const lockpath = fs.existsSync(path.join(c.root, 'package-lock.json')) ?
